nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -1423,9 +1423,40 @@ public static Timestamp toTimestamp(final Object value, final Supplier<DateForma
             }
         }
 
+        if ("oracle.sql.TIMESTAMP".equals(value.getClass().getName())) {
```

Comment:
Introducing vendor-specific type handling in this generic `DataTypeUtils` class seems problematic. Instead of introducing custom code, is it possible format the SQL statement so that the result type is a string that could be parsed using existing configurable Timestamp formatting?

Reply:
Thanks for review.  You are right, and I redesign a new solution for this issue.  
Oracle jdbc driver will tranform oracle.sql..TIMESTAMP to java.sql.Timestamp. 
So it's ok if we just modify ResultSetRecordSet.java